### ISSUE TITLE
keyboard focus for screen reader

### ISSUE THREAD

**Issue by woreda** [UnigramDev/Unigram#1557]
unigram should keep the focus of the keyboard access when the user returns to unigram window after closing and opening a chat.
currently, when unigram is open on a chat the keyboard focus for screen reader NVDA is randomly selected. 
but, either it should go to where it has been previously or it should go to the start of the unread messages.

**Comment by kranid** [comment]
i can confirm this bug too.

**Comment by FrayxRulez** [comment]
Hello, focus is usually moved to the chat text field because that’s usually where most users want their focus to be. 
I was actually thinking that I could introduce a “Optimize app for screen readers”, that would disable focus redirect as a start. What do you think?

**Comment by woreda** [comment]
the problem is not the focus moving to the chat field. it is rather when we tab back to the messages, the message is selected randomly. what is expected is the focus to return to where it has been. and when we return after closing unigram it the focus in the messages should at the start of unread messages if we have been there. and we have scrolled up before, it should remember where we were.
but, if the app could optimize itself when it detects screen reader it is a very good idea.

**Comment by Dyaminigo** [comment]
I agree with @woreda. Focus should be on the last message read or the first unread message. But I cannot but mention the problem when you read the message history, and new messages are actively added to the chat, sometimes the keyboard focuses on the text field, although this should only happen when the chat is opening, and not when is in it.

**Comment by MarcAnt01** [comment]
Why has this been closed?

**Comment by woreda** [comment]
is there any development on this issue?
Can I help to further diagnose the problem?

**Comment by sukiletxe** [comment]
Yesterday this was brought up in Unigram Insiders. Basically the developer thinks he has no way to fix this, as fixing it would break scrolling (or that's what I understood). The ListView he uses to display messages handles focus internally, and changing it would break things.

**Comment by ultrasound1372** [comment]
It has no way of causing a specific item to fire a focus event?? Would this break the visual experience if it were scrolled into focus?

**Comment by sukiletxe** [comment]
If this helps, I've done some minimal testing, and if you press escape on an unread message and then return to it, the unread message is the next one (as it should be). That is to say: focus an unread message which is not the last one, then press escape. When you return, the first unread message is the next one. Thus it remember something having to do about focus.

**Comment by AlexandrKozlovskiy** [comment]
Yes,i as blind user confirm this issue.

**Comment by AlexandrKozlovskiy** [comment]
In my opinion it should be fixed,because people can forgot press escape key,and focus will be loose. Also when we press on link on some message focus not move to this message,but move randomly too. After pressing escape,if the referenced message from other channel or group,focus not move again in unread messages,but move randomly too. I hope @FrayxRulez will pay attention to this issue.

**Comment by FrayxRulez** [comment]
Should be done now
